### ISSUE TITLE
7.9.1 Can't compose any email. No outbound accounts shown in "From" dropdown

### ISSUE THREAD

**Issue by pgorod** [salesagility/SuiteCRM#3741]
This is on a 7.9.1 test instance, upgraded from 7.9 (and from many earlier versions).

#### Steps to Reproduce
1. Configure system outbound email account, in my case Gmail. Click the checkbox to `allow users to use this account`.
2. Send test email (works well)
3. Go to `Emails` module (this is the Admin user, and has no configured Inbound email)
4. Click `Compose`. Note that there is nothing in the `From` dropdown box.
5. Fill in a valid `To` address, fill in `Subject` and `Body`.
6. If you click `Send`, you get an error `Error Sending Email. Please contact your administrator for assistance.`

#### In the logs

suitecrm.log shows this at the moment the `Send` button is pressed:

```
Sun Jun 18 15:12:53 2017 [1470][1][FATAL] SugarPHPMailer encountered an error: Invalid address:  (addAnAddress Reply-To): null
Sun Jun 18 15:12:53 2017 [1470][1][FATAL] SugarPHPMailer encountered an error: Invalid address:  (punyEncode) null
```
#### In the database

This is a small CSV with the only row in my `outbound_email` table:

```
"id","name","type","user_id","mail_sendtype","mail_smtptype","mail_smtpserver","mail_smtpport","mail_smtpuser","mail_smtppass","mail_smtpauth_req","mail_smtpssl","date_entered","date_modified","modified_user_id","created_by","deleted","assigned_user_id"
"b841a42e-7e21-d813-f871-594684657b17","system","system","1","smtp","other","smtp.gmail.com","587","obfuscated@gmail.com","TLOBFUSCATEDsjkdskjfdskjfdsjafcYPA=","1","2","2017-06-18 13:46:24","2017-06-18 13:46:24","1","1","0",NULL
```

#### Environment
* SuiteCRM Version used: 7.9.1
* Browser name and version: Firefox latest, and Edge. Same behavior.
* Environment name and version: MySQL 5.7, PHP 7
* Operating System and version: Ubuntu 16.04

**Comment by pgorod** [comment]
@Dillon-Brown were you able to reproduce this problem? I'm trying to understand if I have a problem specific to my system (and a few other people out there) or if this is happening to everybody. Thanks.

**Comment by Dillon-Brown** [comment]
@pgorod I'm able to replicate this. With no inbound email account configured; the "From" field won't populate.

**Comment by samus-aran** [comment]
Adding a comment reference from @halest
https://github.com/salesagility/SuiteCRM/issues/3790#issuecomment-311624779

**Comment by pgorod** [comment]
@samus-aran I recommend elevating this one to High priority also. Most of the critical problems with email being reported in the forums in 7.9.3 are related to this. 

I am not sure that this issue expresses the full problem of the missing accounts in the "From" field. Maybe we can pick up some more variations of this from the forums.

Maybe some educational effort is also needed to explain the changes in outbound accounts. First of all, for me, since I am a bit confused and it makes my help in the forums less efficient; and also for everybody else, of course. Thanks

**Comment by Dillon-Brown** [comment]
This appears to be incorrectly labelled as a bug. The "From" field won't populate unless the user has a valid inbound and outbound email account.

However, the "From" field failing to populate with a valid outbound and valid group email is definitely a bug #4111

**Comment by pgorod** [comment]
Hi @Dillon-Brown 

at least in terms of user experience, this is still a bug... some proper message should be given to the user, or else it is very confusing. This keeps sending people into the forums asking questions...

**Comment by Dillon-Brown** [comment]
@pgorod Agreed, this should definitely be made clearer.

**Comment by dspaan** [comment]
Just reporting back that i lost a lot of time because of this bug, i do not see why i need to create an inbound account in order to send outbound e-mail. I see this issue has been here since version 7.9.1 and we're on 7.9.4

**Comment by pgorod** [comment]
My past half an hour was spent getting around another instance of this lack of proper messages.

In one of my test systems which I just upgraded to 7.9.7, I couldn't get the "From" box to populate, despite having an inbound email configured. It turns out this same email was defined as a "group email" elsewhere (something I don't remember doing, but probably did). It would be nice to warn users when the "From" box is empty, which accounts were discarded and why.

**Comment by fcorluka** [comment]
Hi guys, 
I see this is closed, but I have similar issue in 7.8.23 LTS.

Do you have any idea how to fix this, or should I open new issue?

**Comment by Dillon-Brown** [comment]
Hi,@fcorluka, It's most likely unrelated to this bug if it's in 7.8.23. The merged fix here was to EmailsController.php which was added in 7.9.x. I'd open a new issue.
